app/views/users/show.html.erb

```diff
@@ -35,6 +35,12 @@
 	<div class="profile-real-name"><%= @user.name %></div>
 </div>
 
-<% if @posts.any? %>
-  <%= render 'gallery' %>
+<% if @user.show_gallery_for?(current_user) %>
```

Comment:
it makes more sense to make `current_user.show_gallery_for?(@user)`

Reply:
also, create `users#current` endpoint where `users#show` will be redirected if it's current user